discovery/azure/azure.go

```diff
@@ -286,6 +287,15 @@ func (d *Discovery) refresh() (tg *targetgroup.Group, err error) {
 				return
 			}
 
+			// We check if the virtual machine has been deallocated/stopped.
+			// If so, we skip them in service discovery.
+			if strings.EqualFold(vm.PowerStateCode, "PowerState/deallocated") ||
+				strings.EqualFold(vm.PowerStateCode, "PowerState/stopped") {
```

Comment:
Can a stopped machine be started again?

For EC2 we still list the stopped machines, as someone may want to monitor them still and leave dropping them to relabelling.